### ISSUE TITLE
calendar.timegm should not be fixed to current time

### ISSUE THREAD

**Issue by fzonneveld** [spulec/freezegun#327]
Getting a calendar.timegm should not use the frozen time, but the time given on input.

**Comment by victornoel** [comment]
I think I'm seeing the same problem, it's happening since 0.3.13, it was working before.

**Comment by pbasista** [comment]
The `timegm` change has been introduced in commit 56712394976b12b5ab5054f4eb4bf09ec6c582e6.

@eric-weaver What was the reason for changing the output of the [`timegm`](https://docs.python.org/3/library/calendar.html#calendar.timegm) function to always respond with a timestamp of the fixed datetime, regardless of the provided input parameter?

**Comment by Ionaru** [comment]
I'm having trouble with the [`python-jwt`](https://pypi.org/project/python-jwt/) plugin and this change.

Both `nbf` and `exp`  use [`timegm`](https://docs.python.org/3/library/calendar.html#calendar.timegm), making them the same value, together with the frozen time set to the exact same time it results in a JWT token that is never valid during tests.

**Comment by itajaja** [comment]
`timegm` is a deterministic helper function and should not be patched by freezegun 🤔

**Comment by ricky-sb** [comment]
> I'm having trouble with the [`python-jwt`](https://pypi.org/project/python-jwt/) plugin and this change.
> 
> Both `nbf` and `exp` use [`timegm`](https://docs.python.org/3/library/calendar.html#calendar.timegm), making them the same value, together with the frozen time set to the exact same time it results in a JWT token that is never valid during tests.

How did you end up resolving this? Monkey-patching `timegm` out of this library?

**Comment by fzonneveld** [comment]
Do not recall this anymore, a long time ago and I am currently not working
anymore at Irdeto, therefore do not have access to the source code to check.
A guess would be that we fixed the version of freezegun in the test
requirements file.
Will forward this to a person within the team I worked to see if he can see
what we did

Regards,
Frank

Op ma 14 dec. 2020 om 21:04 schreef Ricky <notifications@github.com>:
